torchaudio/datasets/quesst14.py

```diff
@@ -0,0 +1,101 @@
+import os
+import re
+from pathlib import Path
+from typing import Tuple, Union, Optional
+
+import torch
+import torchaudio
+from torch.hub import download_url_to_file
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import extract_archive
+
+
+URL = "https://speech.fit.vutbr.cz/files/quesst14Database.tgz"
+_CHECKSUM = "4f869e06bc066bbe9c5dde31dbd3909a0870d70291110ebbb38878dcbc2fc5e4"
+_LANGUAGES = [
+    "albanian",
+    "basque",
+    "czech",
+    "nnenglish",
+    "romanian",
+    "slovak",
+]
+
+
+class QUESST14(Dataset):
```

Comment:
@carolineechen Can we have `Quesst14` as the name of the dataset?
I understand that the original name is all capital but there is no benefit going against PEP-8's naming convention. Currently `torchaudio.datasets` has couple of datasets named like that but that's definitely not a standard and we should not make it a standard.

Reply:
I agree we shouldn't default to all caps for dataset naming as we have done with several datasets in the past, but as this dataset's original name is already all capitalized (["QUESST 2014 Multilingual..."](https://speech.fit.vutbr.cz/software/quesst-2014-multilingual-database-query-by-example-keyword-spotting)), I think it makes sense for this to remain all caps, and to follow the dataset's conventional naming. What do you think?

Reply:
I think this is where the sticking with the PEP-8 naming convention matters.
It is unfortunate that existing datasets violate them and we are not gonna change it for the sake of changing it, but
as we add more datasets, we should stick with PEP-8. 

The reason is that boundary of allowing all caps and not are kinda arbitrary this manner.
This can lead to discussion and disagreement every time we add datasets
 It's like formatting issue, where subjectivity causes disagreement and make the collaboration unproductive.
Sticking with PEP-8 allows us to reason why more clearly. If all caps are required from technical reason, it will be clearer.